app/src/components/Filters/AddFilterButton.tsx

```diff
@@ -1,8 +1,10 @@
 import { Button, HStack, Icon, Text } from "@chakra-ui/react";
 import { BsPlus } from "react-icons/bs";
-import { comparators, useFilters } from "./useFilters";
+import { useFilters } from "./useFilters";
+import { comparators } from "~/types/shared.types";
+import { type FilterOptionType } from "./types";
 
-const AddFilterButton = ({ filterOptions }: { filterOptions: string[] }) => {
+const AddFilterButton = ({ filterOptions }: { filterOptions: FilterOptionType[] }) => {
```

Comment:
Minor, but we could define a helper type like follows:

`type AtLeastOne<T> = [T, ...T[]];`

And then change the type here to `filterOptions: AtLeastOne<FilterOptionType>`.

That way TS would enforce there's at least one filter option passes, so you wouldn't have to cast the `as string` below.

Reply:
Yes, this does save some casting (unfortunately it adds an `unknown` casting in `Filters.tsx`, but at least the reasoning behind the cast is explained by its surrounding code).